internal/controller/controllers/infraenv_controller.go

```diff
@@ -573,8 +578,19 @@ func (r *InfraEnvReconciler) updateInfraEnvStatus(
 		infraEnv.Status.ISODownloadURL = internalInfraEnv.DownloadURL
 		imageCreatedAt := metav1.NewTime(time.Time(internalInfraEnv.GeneratedAt))
 		infraEnv.Status.CreatedTime = &imageCreatedAt
+	}
 
-		// set initrd and script endpoint here so we're not changing the auth token constantly
+	existingInitrdURL, err := url.Parse(infraEnv.Status.BootArtifacts.InitrdURL)
+	if err != nil {
+		return r.handleEnsureISOErrors(ctx, log, infraEnv, err, internalInfraEnv)
+	}
+	// KernelURL is not signed so it gets updated on every reconciliation, thus it has correct URL schema.
+	desiredKernelURL, err := url.Parse(infraEnv.Status.BootArtifacts.KernelURL)
+	if err != nil {
+		return r.handleEnsureISOErrors(ctx, log, infraEnv, err, internalInfraEnv)
+	}
+	if existingInitrdURL.Scheme != desiredKernelURL.Scheme {
```

Comment:
Would it be more clear to use `r.InsecureIPXEURLs` rather than parsing out the previously set artifact?
So something like:
```
if r.InsecureIPXEURLs && existingInitrdURL == "https" || !r.InsecureIPXEURLs && existingInitrdURL == "http" {
```

Alternatively you could break this out into a helper like `func isHttpsURL(url string) bool` if the condition is too ugly.

Reply:
It just feels weird to update just the initrd URL separately. It feels a bit better when we keep kernelURL and InitrdURL schemas in sync

Reply:
> It just feels weird to update just the initrd URL separately.

Isn't this what we're already doing though? The value of `r.InsecureIPXEURLs` is the source of truth for what all the scheme's should be, why introduce an additional dependency between urls?

Reply:
Reverted back to d61b506fec1751e2debdc4bbd70ff6500bc6ef74 - breaks edge-subsystem-kubeapi-aws:
```
* Subsystem Suite: [kube-api]cluster installation verify InfraEnv ISODownloadURL and image CreatedTime are not changing - update Annotations
* Subsystem Suite: [kube-api]cluster installation verify InfraEnv ISODownloadURL and image CreatedTime are changing - update IgnitionConfigOverride
```